addons/binding/org.openhab.binding.cm11a/src/main/java/org/openhab/binding/cm11a/handler/Cm11aApplianceHandler.java

```diff
@@ -0,0 +1,95 @@
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.cm11a.internal.InvalidAddressException;
+import org.openhab.binding.cm11a.internal.X10Interface;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler for Appliance (also called Switch) modules. These modules only support ON and OFF states
+ *
+ * @author Bob Raker
+ *
+ */
+public class Cm11aApplianceHandler extends Cm11aAbstractHandler {
+
+    private Logger logger = LoggerFactory.getLogger(Cm11aApplianceHandler.class);
+
+    private State desiredState = UnDefType.UNDEF;
+
+    public Cm11aApplianceHandler(Thing thing) {
+        super(thing);
+        this.thing = thing;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("**** Cm11aApplianceHandler handleCommand command = {}, channelUID = {}", command.toString(),
+                channelUID.getAsString());
+
+        x10Function = 0;
+        Bridge bridge = getBridge();
+        this.channelUID = channelUID;
+
+        Cm11aBridgeHandler cm11aHandler = (Cm11aBridgeHandler) bridge.getHandler();
+        if (cm11aHandler != null && cm11aHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+            if (command.equals(OnOffType.ON)) {
+                x10Function = X10Interface.FUNC_ON;
+                desiredState = OnOffType.ON;
+            } else if (command.equals(OnOffType.OFF)) {
+                x10Function = X10Interface.FUNC_OFF;
+                desiredState = OnOffType.OFF;
+            } else if (command instanceof RefreshType) {
+                x10Function = X10Interface.FUNC_OFF;
+                desiredState = OnOffType.OFF;
+                logger.info("Received REFRESH command for switch {}", houseUnitCode);
+            }
+
+            if (x10Function > 0) {
+                X10Interface x10Interface = cm11aHandler.getX10Interface();
+                x10Interface.scheduleHWUpdate(this);
+            } else {
+                logger.debug("Received invalid command for switch {} command: {}", houseUnitCode, command.toString());
+            }
+        } else {
+            logger.debug("Attempted to change switch to {} for {} because the cm11a is not online", command.toString(),
+                    houseUnitCode);
+        }
+    }
+
+    @Override
+    public void updateHardware(X10Interface x10Interface) throws IOException, InvalidAddressException {
+        if (!desiredState.equals(currentState)) {
+            try {
+                if (x10Interface.sendFunction(houseUnitCode, x10Function)) {
+                    // Hardware update was successful so update openHAB
+                    updateState(channelUID, desiredState);
+                    setCurrentState(desiredState);
+                }
+            } catch (Exception e) {
```

Comment:
You used to catch two exceptions I suggested to replace it like follow any reason you changed it to this:

```java
} catch (InvalidAddressException | IOException e) {
    logger.error( "cm11a was not able to update the cm11a because of an exception. Check your hardware.", e);
}
```

Please also remove the `throws IOException, InvalidAddressException` and from the interface, please also validate that you do not catch them on places where they will not occur.